### PR TITLE
Travis CI: Run all flake8 checks on modified code

### LINKED ISSUE #2300: Run flake8 checks in Travis on modified code
### Is your feature request related to a problem? Please describe.
We can currently introduce flake8 errors into our code and have no linting/error checking.

### Describe the solution you'd like
This site shows how to run flake8 tests on the modified code. It's in the context of pre-commit hooks, but could we use the code in our Travis build? 
- https://consideratecode.com/2016/10/15/check-code-changes-with-flake8-before-committing/

Code from site:
```sh
git diff --cached -U0 | flake8 --diff
```

It notes that this won't catch _all_ issues, so another check that could be performed would be to perform a full check on master and on the PR, and ensure the outputs are the same (i.e. no new errors).

### Stakeholders
@cclauss @hornc

### PR DESCRIPTION
**Pull request by cclauss**
Closes #2300

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,12 @@ jobs:
     - name: “Python 3.8 make lint only”
       python: "3.8"
       install: pip install flake8
-      script: make lint
+      script:
+        - make lint
+        # Run all flake8 tests on modified code
+        - if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then
```
Comment (cdrini) [inline .travis.yml]: Out of curiosity, why is this needed?

File: .travis.yml [hunk 2]
```diff
@@ -11,7 +11,12 @@ jobs:
...
+            git diff origin/master -U0 | flake8 --diff;
```
Comment (cdrini) [inline .travis.yml]: I'm a little nervous about checking _all_ flake8 errors; but I'm happy to try it to see how many of the open PRs break 😅 .

File: .travis.yml (hunk 1 above)
Comment (cclauss) [reply .travis.yml]: `origin/master` is not defined in a push...
https://travis-ci.org/internetarchive/openlibrary/jobs/649813381#L354

Comment (cdrini) [reply .travis.yml]: Hmm, wait, then when is it running? Cause it should ideally be run every time one pushes to a pull request.

**Comment by cclauss** [comment]
The top on is a `pull_request` and the bottom on is a `push`...
Screenshot 2020-02-14 at 00 24 17

File: .travis.yml (hunk 1 above)
Comment (cdrini) [reply .travis.yml]: Ahhh, ok; maybe let's add a comment here about why the check exists :+1:

**Comment by cclauss** [comment]
What is our `--max-line-length`?  PEP8 say max 79 characters per line but almost no one still believes that is a good approach.  `psi/black` uses 88 characters which I find to be quite sane.  The GitHub editor is 127 characters which is very roomy.

**Comment by cdrini** [comment]
88 seems a little tight, but I'm ok with it; we can adapt :)

**Comment by cdrini** [comment]
Thanks for the 👀 folks! Let's see what breaks :P
